### PR TITLE
Fix #44

### LINKED ISSUE #44: slash command for digression
For this you'll need to edit this [file](https://github.com/pyslackers/sirbot-pythondev/blob/master/sirbot/pythondev/slack.py) and also the [configuration file](https://github.com/pyslackers/sirbot-pythondev/blob/master/sirbot.yml) to put the static string in.

You can take example on the `/do` command.

### PR DESCRIPTION
**Pull request by ngrjs**
Partial fix for digression warning. Working as a message, need to convert it to slash command.

### CODE REVIEW

**Comment by ovv** [comment]
Working great @ngrjs 

I can merge or do you want to wait to convert it to a slash command ?

**Comment by ngrjs** [comment]
I wanted to convert that into slash command. But, I haven't been able to make slash commands working at my end.

**Comment by ovv** [comment]
Oh yeah forgot about that bit. 

Could you open an issue in pyslackers/sirbot-slack with your configuration file, what you do with ngrok and the log of sirbot ?

Thanks

**Comment by ngrjs** [comment]
Yup 👍

File: sirbot.yml [hunk 1]
```diff
@@ -82,6 +82,9 @@ pythondev:
     seems so professional!\n\n
 
     May your :taco:s be plentiful!
+  moveto:
+    msg: >
+        Looks like conversation is going off topic. Please move the conversation to {}
```
Comment (ovv) [inline sirbot.yml]: Not the same text as in `test.yml`. I guess the correct formatting is the other one

File: sirbot/pythondev/slack.py [hunk 2]
```diff
@@ -211,6 +213,7 @@ def add(self, slack):
             return
 
     async def leaderboard(self, command, slack, registry):
+        print("here")
```
Comment (ovv) [inline sirbot/pythondev/slack.py]: You should delete that

File: sirbot/pythondev/slack.py [hunk 3]
```diff
@@ -454,3 +457,20 @@ def add(self, slack):
             message.text = message.text.format(number=len(members),
                                                user=event['user']['id'])
             await slack.send(message)
+
+    async def move_to(self, command, slack, registry):
+        response = command.response()
+        to_id = command.text
+        if not to_id:
+            response.text = 'Please enter the channel name'
+        if to_id.startswith('#'):
```
Comment (ovv) [inline sirbot/pythondev/slack.py]: `elif`

File: sirbot/pythondev/slack.py [hunk 4]
```diff
@@ -454,3 +457,20 @@ def add(self, slack):
...
+            to_id = to_id.replace('#', '')
```
Comment (ovv) [inline sirbot/pythondev/slack.py]: you might want to add `.strip()` too in case someone put a space.

File: sirbot/pythondev/slack.py [hunk 5]
```diff
@@ -454,3 +457,20 @@ def add(self, slack):
             message.text = message.text.format(number=len(members),
                                                user=event['user']['id'])
             await slack.send(message)
+
+    async def move_to(self, command, slack, registry):
+        response = command.response()
+        to_id = command.text
```
Comment (ovv) [inline sirbot/pythondev/slack.py]: you should rename `to_id` to something more explicit. It's not the `id` but the `name`

File: test.yml [hunk 6]
```diff
@@ -79,6 +79,9 @@ pythondev:
     seems so professional!\n\n
 
     May your :taco:s be plentiful!
+  moveto:
+    msg: >
+        Looks like conversation is going off topic. Please move the conversation to <#{}|channel>
```
Comment (ovv) [inline test.yml]: I believe `channel` should be the actual channel name (`to.name`)

**Review by ovv — CHANGES_REQUESTED** [review]
:+1: Nice work

Just some small tweaks and one ting to change but I need to test to remember exactly how it works. The idea being that we don't want everyone to see the errors message `Sorry I can not understand the destination.`. There is a way to display them only to the person triggering the command. I'll let you know how to do it.

**Comment by ovv** [comment]
Hey @ngrjs 

I made some update to the slack commands API today to try to makes them easier. By default when using `command.response()` the response will be `in_channel` meaning that everybody will be able to see it. To change that you can pass `ephemeral` as the `type_` argument and the message will be only displayed for the user using the command.

this can be changed later by `reponse.response_type = 'ephemeral'`

Due to the changes the `public` argument to the `add_command` as been deleted.

I updated other commands so you can take a look :)

**Pull request closed without merging**